### PR TITLE
Saulparedes/genpolicy sync [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

- [ ] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
  - [ ] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [ ] Aware about the PR to be merged using "create a merge commit" rather than "squash and merge" (or similar)
- [ ] genPolicy only: Ensured the tool still builds on Windows
- [ ] genPolicy only: Updated sample YAMLs' policy annotations, if applicable
- [ ] The `upstream-missing` label (or `upstream-not-needed`) has been set on the PR.

### CODE REVIEW

**Comment by Redent0r** [comment]
Do not merge until changes in https://github.com/kata-containers/kata-containers/pull/9185/commits are incorporated

File: src/tools/genpolicy/genpolicy-advanced-command-line-parameters.md [hunk 1]
```diff
@@ -83,10 +83,10 @@ $ genpolicy -j my-settings.json -y test.yaml
 
 # Use a custom path to `genpolicy` input files
 
-By default, the `genpolicy` input files [`rules.rego`](rules.rego) and [`genpolicy-settings.json`](genpolicy-settings.json) must be present in the current directory - otherwise `genpolicy` returns an error. Users can specify a different path to these two files, using the `-i` command line parameter - e.g.,
+By default, the `genpolicy` input files [`rules.rego`](rules.rego) and [`genpolicy-settings.json`](genpolicy-settings.json) must be present in the current directory - otherwise `genpolicy` returns an error. Users can specify different paths to these two files, using the `-p` and `-j` command line parameters - e.g.,
```
Comment (danmihai1) [inline src/tools/genpolicy/genpolicy-advanced-command-line-parameters.md:86]: This breaking change requires a corresponding change in az confcom. Seth didn't mind making this change several weeks ago, but we need to sync up with him again.

File: src/tools/genpolicy/genpolicy-advanced-command-line-parameters.md [hunk 2]
```diff
@@ -83,10 +83,10 @@ $ genpolicy -j my-settings.json -y test.yaml
...
 
 ```bash
-$ genpolicy -i /tmp -y test.yaml
+$ genpolicy -p /tmp/rules.rego -j /tmp/genpolicy-settings.json -y test.yaml
```
Comment (danmihai1) [inline src/tools/genpolicy/genpolicy-advanced-command-line-parameters.md:89]: Do we need to update MSFT tests for this?

File: src/tools/genpolicy/genpolicy-settings.json [hunk 3]
```diff
@@ -136,6 +136,17 @@
     },
     "volumes": {
         "emptyDir": {
+            "mount_type": "local",
+            "mount_source": "^$(cpath)/$(sandbox-id)/rootfs/local/",
```
Comment (danmihai1) [inline src/tools/genpolicy/genpolicy-settings.json:140]: These paths might need to change when using confidential=true at MSFT. They are probably specific to main code on Mariner, *without* confidential enabled.

File: src/tools/genpolicy/genpolicy-settings.json [hunk 4]
```diff
@@ -277,7 +288,10 @@
         ]
     },
     "kata_config": {
-        "confidential_guest": true
+        "confidential_guest": false
```
Comment (danmihai1) [inline src/tools/genpolicy/genpolicy-settings.json:291]: I believe MSFT needs this to be true.

File: src/tools/genpolicy/src/config_map.rs [hunk 5]
```diff
@@ -116,10 +116,6 @@ impl yaml::K8sResource for ConfigMap {
         panic!("Unsupported");
     }
 
-    fn get_namespace(&self) -> String {
```
Comment (danmihai1) [inline src/tools/genpolicy/src/config_map.rs:119]: Removing this code seems suspect at a quick look.

**Review by danmihai1 — CHANGES_REQUESTED** [review]
I think you should split this PR in smaller pieces, that are easier to review, test, debug.

**Comment by Redent0r** [comment]
Closing for https://github.com/microsoft/kata-containers/pull/171

**Pull request closed without merging**